serverless/src/layer.ts

```diff
@@ -228,6 +226,14 @@ export function applyLayers(
       lambdaLibraryLayerArn = getLambdaLibraryLayerArn(region, javaLayerVersion, lambda.runtime, lambda.architecture);
       addLayer(lambdaLibraryLayerArn, lambda);
     }
+  });
+  return errors;
+}
+
+export function applyExtensionLayer(region: string, lambdas: LambdaFunction[], extensionLayerVersion?: number) {
+  const errors: string[] = [];
+  lambdas.forEach((lambda) => {
+    let lambdaExtensionLayerArn;
 
     if (extensionLayerVersion !== undefined) {
```

Comment:
I feel like we should do here similarly to what we do above. If the extension version is not set, return an error. Otherwise, I think it will be real confusing for people.

Reply:
I have added this functionality and tested it.